core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -417,6 +417,8 @@ class SparkContext(config: SparkConf) extends Logging {
     if (!_conf.contains("spark.app.name")) {
       throw new SparkException("An application name must be set in your configuration")
     }
+    // HADOOP-19097 Set fs.s3a.connection.establish.timeout to 30s
+    conf.setIfMissing("spark.hadoop.fs.s3a.connection.establish.timeout", "30s")
```

Comment:
Can we add a comment that this can be removed after Hadoop 3.4.1?

Reply:
Sure! Thank you for review.